lab_4_retrieval_w_clustering/main.py

```diff
@@ -271,11 +375,21 @@ def __init__(self, db: DocumentVectorDB, n_clusters: int) -> None:
             db (DocumentVectorDB): An instance of DocumentVectorDB class.
             n_clusters (int): Number of clusters.
         """
+        self.__clusters = []
+        self._db = db
+        self._n_clusters = n_clusters
 
     def train(self) -> None:
         """
         Train k-means algorithm.
         """
+        initial_centroids = self._db.get_vectors()[:self._n_clusters]
+        self.__clusters = [ClusterDTO(centroid[1]) for centroid in initial_centroids]
+        self.run_single_train_iteration()
+        while True:
+            self.run_single_train_iteration()
+            if self._is_convergence_reached(self.__clusters):
```

Comment:
looks like you compare `self.__clusters` with `self.__clusters`. Are tests green??

Reply:
i understand why you asked that, but `_is_convergence_reached` receives new_clusters as input, so i don't know how to compare old and new clusters with these conditions. i tried saving the clusters before `run_single_train_iteration` and passing them to the function, but it didn't change anything
and my tests are very green

Reply:
i tried this version:
`while True:
            old_clusters = copy.deepcopy(self.__clusters)
            self.run_single_train_iteration()
            if self._is_convergence_reached(old_clusters):
                break`